message_sender/views.py

```diff
@@ -138,6 +146,22 @@ def create(self, request, *args, **kwargs):
         else:
             channel = Channel.objects.get(channel_id=kwargs.get('channel_id'))
 
+        if request.data.get("channel_data", {}).get("session_event", None) == \
+                "close":
+            msisdn = request.data.pop("from")
+        else:
+            msisdn = request.data.pop("from_addr")
```

Comment:
I'm a bit unsure what we are doing here. Are we using this if statement to determine if the message is from vumi or from Junebug? If so, then this if statement will only ever validate Junebug messages that are session close events.

A better way would be just to check if there's a `from` value, and if there is, use that for the msisdn, else use `from_addr`.